wont build in linux mint 22.1 when it has before

I am using linux mint 22.1 (T2-Linux - https://github.com/t2linux/T2-Mint)

I manage to to compile mangohud in past before but wanted to start fresh and when I try compile i get errors and fails somewhere around dbus part and and i checked for all dependency needed and i have them all.

here is output from the terminal:
```

# Checking dependencies for "linuxmint"
# Unable to find distro information!
# Attempting to build regardless
# Checking dependencies for "ubuntu"
The Meson build system
Version: 1.3.2
Source dir: /tmp/MH/MangoHud
Build dir: /tmp/MH/MangoHud/build/meson64
Build type: native build
Project name: MangoHud
Project version: v0.8.1
C compiler for the host machine: cc (gcc 13.3.0 "cc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
C linker for the host machine: cc ld.bfd 2.42
C++ compiler for the host machine: c++ (gcc 13.3.0 "c++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
C++ linker for the host machine: c++ ld.bfd 2.42
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python3 (mako) found: YES (/usr/bin/python3) modules: mako
Checking if "GCC atomic builtins" compiles: YES 
Checking if "Supports timespec_get" compiles: YES 
Checking for function "bswap32" : YES 
Checking for function "bswap64" : YES 
Checking for function "clz" : YES 
Checking for function "clzll" : YES 
Checking for function "ctz" : YES 
Checking for function "expect" : YES 
Checking for function "ffs" : YES 
Checking for function "ffsll" : YES 
Checking for function "popcount" : YES 
Checking for function "popcountll" : YES 
Checking for function "unreachable" : YES 
Found pkg-config: YES (/usr/bin/pkg-config) 1.8.1
Run-time dependency x11 found: YES 1.8.7
Run-time dependency wayland-client found: YES 1.23.0
Run-time dependency dbus-1 found: YES 1.14.10
Run-time dependency xkbcommon found: YES 1.6.0
Run-time dependency threads found: YES
Compiler for C supports arguments -Werror=implicit-function-declaration: YES 
Compiler for C supports arguments -Werror=missing-declarations: YES 
Compiler for C supports arguments -Werror=missing-prototypes: YES 
Compiler for C supports arguments -Werror=return-type: YES 
Compiler for C supports arguments -Werror=incompatible-pointer-types: YES 
Compiler for C supports arguments -Wno-unused-parameter: YES 
Compiler for C supports arguments -Qunused-arguments: NO 
Compiler for C supports arguments -fno-math-errno: YES 
Compiler for C supports arguments -fno-trapping-math: YES 
Compiler for C supports arguments -Wno-missing-field-initializers: YES 
Compiler for C++ supports arguments -Werror=missing-declarations: YES 
Compiler for C++ supports arguments -Werror=return-type: YES 
Compiler for C++ supports arguments -Wno-unused-parameter: YES 
Compiler for C++ supports arguments -Qunused-arguments: NO 
Compiler for C++ supports arguments -fno-math-errno: YES 
Compiler for C++ supports arguments -fno-trapping-math: YES 
Compiler for C++ supports arguments -Wno-non-virtual-dtor: YES 
Compiler for C++ supports arguments -Wno-missing-field-initializers: YES 
Checking for function "dlopen" : YES 
Checking for function "clock_gettime" : YES 
Downloading vulkan-headers source from https://github.com/KhronosGroup/Vulkan-Headers/archive/v1.2.158.tar.gz
Downloading file of unknown size.
Downloading vulkan-headers patch from https://wrapdb.mesonbuild.com/v2/vulkan-headers_1.2.158-2/get_patch
Download size: 1107
Downloading: ..........

Executing subproject vulkan-headers 

vulkan-headers| Project name: vulkan-headers
vulkan-headers| Project version: 1.2.158
vulkan-headers| C compiler for the host machine: cc (gcc 13.3.0 "cc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
vulkan-headers| C linker for the host machine: cc ld.bfd 2.42
vulkan-headers| Build targets in project: 1
vulkan-headers| Subproject vulkan-headers finished.

Checking for size of "void*" : 8 
Found CMake: /usr/bin/cmake (3.28.3)
Run-time dependency imgui found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency imgui
Downloading imgui source from https://github.com/ocornut/imgui/archive/refs/tags/v1.89.9.tar.gz
Downloading file of unknown size.
Downloading imgui patch from https://wrapdb.mesonbuild.com/v2/imgui_1.89.9-1/get_patch
Download size: 2413
Downloading: ..........

Executing subproject imgui 

imgui| Project name: imgui
imgui| Project version: 1.89.9
imgui| C++ compiler for the host machine: c++ (gcc 13.3.0 "c++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
imgui| C++ linker for the host machine: c++ ld.bfd 2.42
imgui| Library d3dcompiler found: NO
imgui| Library d3d9 skipped: feature dx9 disabled
imgui| Library d3d10 skipped: feature dx10 disabled
imgui| Library d3d11 skipped: feature dx11 disabled
imgui| Library d3d12 skipped: feature dx12 disabled
imgui| Dependency appleframeworks (modules: Foundation, AppKit, GameController, Metal) skipped: feature metal disabled
imgui| Dependency gl skipped: feature opengl disabled
imgui| Dependency sdl2 skipped: feature sdl_renderer disabled
imgui| Dependency vulkan skipped: feature vulkan disabled
imgui| Has header "webgpu/webgpu.h" skipped: feature webgpu disabled
imgui| Dependency glfw3 skipped: feature glfw disabled
imgui| Dependency sdl2 skipped: feature sdl2 disabled
imgui| Dependency appleframeworks (modules: Carbon, Cocoa, GameController) skipped: feature osx disabled
imgui| Library dwmapi skipped: feature win disabled
imgui| Dependency allegro-5 skipped: feature allegro5 disabled
imgui| Dependency allegro_primitives-5 skipped: feature allegro5 disabled
imgui| Build targets in project: 3
imgui| Subproject imgui finished.

Dependency imgui from subproject subprojects/imgui-1.89.9 found: YES 1.89.9
Run-time dependency implot found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency implot
Downloading implot source from https://github.com/epezent/implot/archive/refs/tags/v0.16.zip
Downloading file of unknown size.
Downloading implot patch from https://wrapdb.mesonbuild.com/v2/implot_0.16-1/get_patch
Download size: 1226
Downloading: ..........

Executing subproject implot 

implot| Project name: ImGui Plotting Library
implot| Project version: 0.16
implot| C++ compiler for the host machine: c++ (gcc 13.3.0 "c++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
implot| C++ linker for the host machine: c++ ld.bfd 2.42
implot| Build targets in project: 4
implot| Subproject implot finished.

Dependency implot from subproject subprojects/implot-0.16 found: YES 0.16
Run-time dependency spdlog found: NO (tried pkgconfig and cmake)
Downloading spdlog source from https://github.com/gabime/spdlog/archive/refs/tags/v1.14.1.tar.gz
Download size: 270896
Downloading: ..........
Downloading spdlog patch from https://wrapdb.mesonbuild.com/v2/spdlog_1.14.1-1/get_patch
Download size: 2965
Downloading: ..........

Executing subproject spdlog 

spdlog| Project name: spdlog
spdlog| Project version: 1.14.1
spdlog| C++ compiler for the host machine: c++ (gcc 13.3.0 "c++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
spdlog| C++ linker for the host machine: c++ ld.bfd 2.42
spdlog| Dependency threads found: YES unknown (cached)
spdlog| Header "format" has symbol "__cpp_lib_format" skipped: feature std_format disabled
spdlog| Dependency fmt skipped: feature external_fmt disabled
spdlog| Build targets in project: 5
spdlog| Subproject spdlog finished.

Program glslang found: YES (/usr/bin/glslang)
Program git found: YES (/usr/bin/git)
Has header "NVCtrl/NVCtrl.h" : YES 
Compiler for C supports link arguments -Wl,-Bsymbolic-functions: YES 
Compiler for C supports link arguments -Wl,-z,relro: YES 
Compiler for C supports link arguments -Wl,--exclude-libs,ALL: YES 
Compiler for C supports link arguments -lGL: YES 
Compiler for C supports link arguments -static-libstdc++: YES 
WARNING: extract_all_objects called without setting recursive
keyword argument. Meson currently defaults to
non-recursive to maintain backward compatibility but
the default will be changed in the future.

Configuring MangoHud.x86_64.json using configuration
Configuring mangohud using configuration
Program appstreamcli found: YES (/usr/bin/appstreamcli)
Build targets in project: 11

MangoHud v0.8.1

  Subprojects
    imgui                 : YES
    implot                : YES
    spdlog                : YES
    vulkan-headers        : YES

  User defined options
    libdir                : lib/mangohud/lib64
    prefix                : /usr
    werror                : true
    append_libdir_mangohud: false

Found ninja-1.11.1 at /usr/bin/ninja
WARNING: Running the setup command as `meson [options]` instead of `meson setup [options]` is ambiguous and deprecated.
The Meson build system
Version: 1.3.2
Source dir: /tmp/MH/MangoHud
Build dir: /tmp/MH/MangoHud/build/meson32
Build type: native build
Project name: MangoHud
Project version: v0.8.1
C compiler for the host machine: gcc -m32 (gcc 13.3.0 "gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
C linker for the host machine: gcc -m32 ld.bfd 2.42
C++ compiler for the host machine: g++ -m32 (gcc 13.3.0 "g++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
C++ linker for the host machine: g++ -m32 ld.bfd 2.42
Host machine cpu family: x86
Host machine cpu: i686
Program python3 (mako) found: YES (/usr/bin/python3) modules: mako
Checking if "GCC atomic builtins" compiles: YES 
Checking if "Supports timespec_get" compiles: YES 
Checking for function "bswap32" : YES 
Checking for function "bswap64" : YES 
Checking for function "clz" : YES 
Checking for function "clzll" : YES 
Checking for function "ctz" : YES 
Checking for function "expect" : YES 
Checking for function "ffs" : YES 
Checking for function "ffsll" : YES 
Checking for function "popcount" : YES 
Checking for function "popcountll" : YES 
Checking for function "unreachable" : YES 
Found pkg-config: YES (/usr/bin/pkg-config) 1.8.1
Run-time dependency x11 found: YES 1.8.7
Run-time dependency wayland-client found: YES 1.22.0
Run-time dependency dbus-1 found: YES 1.14.10
Run-time dependency xkbcommon found: YES 1.6.0
Run-time dependency threads found: YES
Compiler for C supports arguments -Werror=implicit-function-declaration: YES 
Compiler for C supports arguments -Werror=missing-declarations: YES 
Compiler for C supports arguments -Werror=missing-prototypes: YES 
Compiler for C supports arguments -Werror=return-type: YES 
Compiler for C supports arguments -Werror=incompatible-pointer-types: YES 
Compiler for C supports arguments -Wno-unused-parameter: YES 
Compiler for C supports arguments -Qunused-arguments: NO 
Compiler for C supports arguments -fno-math-errno: YES 
Compiler for C supports arguments -fno-trapping-math: YES 
Compiler for C supports arguments -Wno-missing-field-initializers: YES 
Compiler for C++ supports arguments -Werror=missing-declarations: YES 
Compiler for C++ supports arguments -Werror=return-type: YES 
Compiler for C++ supports arguments -Wno-unused-parameter: YES 
Compiler for C++ supports arguments -Qunused-arguments: NO 
Compiler for C++ supports arguments -fno-math-errno: YES 
Compiler for C++ supports arguments -fno-trapping-math: YES 
Compiler for C++ supports arguments -Wno-non-virtual-dtor: YES 
Compiler for C++ supports arguments -Wno-missing-field-initializers: YES 
Checking for function "dlopen" : YES 
Checking for function "clock_gettime" : YES 

Executing subproject vulkan-headers 

vulkan-headers| Project name: vulkan-headers
vulkan-headers| Project version: 1.2.158
vulkan-headers| C compiler for the host machine: gcc -m32 (gcc 13.3.0 "gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
vulkan-headers| C linker for the host machine: gcc -m32 ld.bfd 2.42
vulkan-headers| Build targets in project: 1
vulkan-headers| Subproject vulkan-headers finished.

Checking for size of "void*" : 4 
Found CMake: /usr/bin/cmake (3.28.3)
Run-time dependency imgui found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency imgui

Executing subproject imgui 

imgui| Project name: imgui
imgui| Project version: 1.89.9
imgui| C++ compiler for the host machine: g++ -m32 (gcc 13.3.0 "g++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
imgui| C++ linker for the host machine: g++ -m32 ld.bfd 2.42
imgui| Library d3dcompiler found: NO
imgui| Library d3d9 skipped: feature dx9 disabled
imgui| Library d3d10 skipped: feature dx10 disabled
imgui| Library d3d11 skipped: feature dx11 disabled
imgui| Library d3d12 skipped: feature dx12 disabled
imgui| Dependency appleframeworks (modules: Foundation, AppKit, GameController, Metal) skipped: feature metal disabled
imgui| Dependency gl skipped: feature opengl disabled
imgui| Dependency sdl2 skipped: feature sdl_renderer disabled
imgui| Dependency vulkan skipped: feature vulkan disabled
imgui| Has header "webgpu/webgpu.h" skipped: feature webgpu disabled
imgui| Dependency glfw3 skipped: feature glfw disabled
imgui| Dependency sdl2 skipped: feature sdl2 disabled
imgui| Dependency appleframeworks (modules: Carbon, Cocoa, GameController) skipped: feature osx disabled
imgui| Library dwmapi skipped: feature win disabled
imgui| Dependency allegro-5 skipped: feature allegro5 disabled
imgui| Dependency allegro_primitives-5 skipped: feature allegro5 disabled
imgui| Build targets in project: 3
imgui| Subproject imgui finished.

Dependency imgui from subproject subprojects/imgui-1.89.9 found: YES 1.89.9
Run-time dependency implot found: NO (tried pkgconfig and cmake)
Looking for a fallback subproject for the dependency implot

Executing subproject implot 

implot| Project name: ImGui Plotting Library
implot| Project version: 0.16
implot| C++ compiler for the host machine: g++ -m32 (gcc 13.3.0 "g++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
implot| C++ linker for the host machine: g++ -m32 ld.bfd 2.42
implot| Build targets in project: 4
implot| Subproject implot finished.

Dependency implot from subproject subprojects/implot-0.16 found: YES 0.16
Run-time dependency spdlog found: NO (tried pkgconfig and cmake)

Executing subproject spdlog 

spdlog| Project name: spdlog
spdlog| Project version: 1.14.1
spdlog| C++ compiler for the host machine: g++ -m32 (gcc 13.3.0 "g++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0")
spdlog| C++ linker for the host machine: g++ -m32 ld.bfd 2.42
spdlog| Dependency threads found: YES unknown (cached)
spdlog| Header "format" has symbol "__cpp_lib_format" skipped: feature std_format disabled
spdlog| Dependency fmt skipped: feature external_fmt disabled
spdlog| Build targets in project: 5
spdlog| Subproject spdlog finished.

Program glslang found: YES (/usr/bin/glslang)
Program git found: YES (/usr/bin/git)
Has header "NVCtrl/NVCtrl.h" : YES 
Compiler for C supports link arguments -Wl,-Bsymbolic-functions: YES 
Compiler for C supports link arguments -Wl,-z,relro: YES 
Compiler for C supports link arguments -Wl,--exclude-libs,ALL: YES 
Compiler for C supports link arguments -lGL: NO 
Compiler for C supports link arguments -static-libstdc++: YES 
WARNING: extract_all_objects called without setting recursive
keyword argument. Meson currently defaults to
non-recursive to maintain backward compatibility but
the default will be changed in the future.

Configuring MangoHud.x86.json using configuration
Configuring mangohud using configuration
Program appstreamcli found: YES (/usr/bin/appstreamcli)
Build targets in project: 11

MangoHud v0.8.1

  Subprojects
    imgui                 : YES
    implot                : YES
    spdlog                : YES
    vulkan-headers        : YES

  User defined options
    libdir                : lib/mangohud/lib32
    prefix                : /usr
    werror                : true
    append_libdir_mangohud: false

Found ninja-1.11.1 at /usr/bin/ninja
WARNING: Running the setup command as `meson [options]` instead of `meson setup [options]` is ambiguous and deprecated.
ninja: Entering directory `build/meson64'
[9/110] Generating src/overlay.frag.spv.h with a custom command
../../src/overlay.frag
[10/110] Generating src/overlay.vert.spv.h with a custom command
../../src/overlay.vert
[51/110] Compiling C++ object src/libMangoHud.a.p/ftrace.cpp.o
FAILED: src/libMangoHud.a.p/ftrace.cpp.o 
c++ -Isrc/libMangoHud.a.p -Isrc -I../../src -I../../include -Isubprojects/imgui-1.89.9 -I../../subprojects/imgui-1.89.9 -I../../subprojects/imgui-1.89.9/backends -I../../subprojects/spdlog-1.14.1/include -I../../subprojects/Vulkan-Headers-1.2.158/include -Isubprojects/implot-0.16 -I../../subprojects/implot-0.16 -I. -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -fvisibility=hidden -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c++14 -O3 -Werror=missing-declarations -Werror=return-type -Wno-unused-parameter -fno-math-errno -fno-trapping-math -Wno-non-virtual-dtor -Wno-missing-field-initializers -fPIC -pthread '-DIMGUI_API=__attribute__((visibility("default")))' -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS '-DPACKAGE_VERSION="v0.8.1"' -DSPDLOG_COMPILED_LIB -DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_DEBUG -D_GNU_SOURCE -DHAVE_PTHREAD -DHAVE_FTRACE -DUSE_GCC_ATOMIC_BUILTINS -DHAVE_TIMESPEC_GET -DHAVE___BUILTIN_BSWAP32 -DHAVE___BUILTIN_BSWAP64 -DHAVE___BUILTIN_CLZ -DHAVE___BUILTIN_CLZLL -DHAVE___BUILTIN_CTZ -DHAVE___BUILTIN_EXPECT -DHAVE___BUILTIN_FFS -DHAVE___BUILTIN_FFSLL -DHAVE___BUILTIN_POPCOUNT -DHAVE___BUILTIN_POPCOUNTLL -DHAVE___BUILTIN_UNREACHABLE '-DMANGOHUD_ARCH="64bit"' -DHAVE_NVML -DNVML_NO_UNVERSIONED_FUNC_DEFS -DHAVE_XNVCTRL -DHAVE_X11 -DHAVE_WAYLAND -DHAVE_DBUS -DVK_USE_PLATFORM_XLIB_KHR -DVK_USE_PLATFORM_WAYLAND_KHR -MD -MQ src/libMangoHud.a.p/ftrace.cpp.o -MF src/libMangoHud.a.p/ftrace.cpp.o.d -o src/libMangoHud.a.p/ftrace.cpp.o -c ../../src/ftrace.cpp
In file included from /usr/include/features.h:502,
                 from /usr/include/x86_64-linux-gnu/c++/13/bits/os_defines.h:39,
                 from /usr/include/x86_64-linux-gnu/c++/13/bits/c++config.h:679,
                 from /usr/include/c++/13/initializer_list:39,
                 from /usr/include/c++/13/array:39,
                 from ../../src/ftrace.h:5,
                 from ../../src/ftrace.cpp:1:
In function ‘ssize_t read(int, void*, size_t)’,
    inlined from ‘void FTrace::FTrace::ftrace_thread()’ at ../../src/ftrace.cpp:83:33:
/usr/include/x86_64-linux-gnu/bits/unistd.h:28:10: error: ‘ssize_t __read_alias(int, void*, size_t)’ specified size between 18446744069414588417 and 18446744073709551611 exceeds maximum object size 9223372036854775807 [-Werror=stringop-overflow=]
   28 |   return __glibc_fortify (read, __nbytes, sizeof (char),
      |          ^~~~~~~~~~~~~~~
/usr/include/c++/13/array: In member function ‘void FTrace::FTrace::ftrace_thread()’:
/usr/include/c++/13/array:109:55: note: destination object allocated here
  109 |       typename __array_traits<_Tp, _Nm>::_Type        _M_elems;
      |                                                       ^~~~~~~~
/usr/include/x86_64-linux-gnu/bits/unistd-decl.h:29:16: note: in a call to function ‘ssize_t __read_alias(int, void*, size_t)’ declared with attribute ‘access (write_only, 2, 3)’
   29 | extern ssize_t __REDIRECT_FORTIFY (__read_alias, (int __fd, void *__buf,
      |                ^~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
[60/110] Compiling C++ object subprojects/implot-0.16/libimplot.a.p/implot_items.cpp.o
ninja: build stopped: subcommand failed.
```

--- a/src/ftrace.cpp
+++ b/src/ftrace.cpp
@@ -62,7 +62,7 @@ void FTrace::ftrace_thread()
 {
     struct {
         std::array<char, 4096> data;
-        unsigned size { 0 };
+        size_t size { 0 };
     } buffer;
 
     while (!stop_thread) {
@@ -85,7 +85,7 @@ void FTrace::ftrace_thread()
             SPDLOG_ERROR("FTrace: reading from trace_pipe failed: {}", strerror(errno));
             break;
         }
-        buffer.size += unsigned(size_read);
+        buffer.size += size_t(size_read);
 
         {
             char *it = buffer.data.begin();

